hyperspy/model.py

```diff
@@ -1087,27 +532,27 @@ def _errfunc4mpfit(self, p, fjac=None, x=None, y=None,
             return [0, self._jacobian(p, y).T]
 
     def _calculate_chisq(self):
-        if self.spectrum.metadata.has_item('Signal.Noise_properties.variance'):
+        if self.signal.metadata.has_item('Signal.Noise_properties.variance'):
 
-            variance = self.spectrum.metadata.Signal.Noise_properties.variance
+            variance = self.signal.metadata.Signal.Noise_properties.variance
             if isinstance(variance, Signal):
                 variance = variance.data.__getitem__(
                     self.axes_manager._getitem_tuple)[self.channel_switches]
         else:
             variance = 1.0
-        d = self(onlyactive=True) - self.spectrum()[self.channel_switches]
+        d = self(onlyactive=True) - self.signal()[self.channel_switches]
```

Comment:
Yes, that's something I forgot to mention - in `Model2D` you set the default value for `channel_switches` as `None`. In numpy that adds a new dimension, which is not what I suspect you want.  Instead, if you want just a generic thing that does not do anything, you should use `slice(None)`:
```python
In [8]: a.shape
Out[8]: (5, 7)

In [9]: a[None].shape
Out[9]: (1, 5, 7)

In [10]: a[slice(None)].shape
Out[10]: (5, 7)
```

Ideally (i.e. that is something we will have in the end, I have no doubts) `channel_switches` for 2d model should be initialized as `self.channel_switches = np.ones(self.xaxis.shape, dtype=bool)` - i.e a 2D array of the "signal" shape